File: estudos/matrix/AbaixoDaMedia.cpp
#include <iostream>
#include <iomanip>

double mean(double * v, int size) {
    double sum = 0;
    for (int i = 0; i < size; i++) {
        sum += v[i];
    }

    return sum / size;
}

int main() {
    int n;
    std::cin >> n;
    
    double grades[3][n];

    for (int i = 0; i < n; i++) {
        std::cin >> grades[0][i];
        std::cin >> grades[1][i];
        std::cin >> grades[2][i];
    }

    for (int i = 0; i < 3; i++) {
        double m = mean(grades[i], n);
        int qtdBellowMean = 0;

        for (int j = 0; j < n; j++) {
            if (grades[i][j] < m) {
                qtdBellowMean++;
            }
        }

        std::cout << "Na " << (i+1) << "a. atividade " << qtdBellowMean << " alunos tiveram nota abaixo da media." << std::endl;
    }
}

File: estudos/matrix/AbaixoDaMedia.java
import java.util.Locale;
import java.util.Scanner;

public class AbaixoDaMedia {
    public static void main(String[] args) {
        // Use Locale.US to ensure the scanner recognizes '.' as a decimal separator
        Scanner scan = new Scanner(System.in).useLocale(Locale.US);
        int n = scan.nextInt();
        double[][] grades = new double[3][n];

        for (int i = 0; i < n; i++) {
            grades[0][i] = scan.nextDouble();
            grades[1][i] = scan.nextDouble();
            grades[2][i] = scan.nextDouble();
        }

        for (int i = 0; i < 3; i++) {
            int qtdBelow = 0;
            double m = AbaixoDaMedia.mean(grades[i]);

            for (int j = 0; j < n; j++) {
                if (grades[i][j] < m) {
                    qtdBelow++;
                }
            }

            System.out.println("Na " + (i+1) + "a. atividade " + qtdBelow + " alunos tiveram nota abaixo da media.");
        }

        scan.close();
    }

    private static double mean(double[] v) {
        double sum = 0;

        for (int i=0; i < v.length; i++) {
            sum += v[i];
        }

        return sum / v.length;
    }
}